frontend/src/pages/add-blog.tsx

```diff
@@ -182,6 +211,79 @@ function AddBlog() {
           Create Blog
         </button>
       </form>
+      {modal && ( //MODAL STARTS
+        <div className="relative z-10"
```

Comment:
Please create a new component and use it here. Helps us in maintaining the code in long run , see which state management helps, for now you can also pass state down the prop, if you feel it very challenging to setup the state by yourself, you can pass handleImageSelect and setFormData in the Modal component and set it there, you can explore little on this part, the good practice would be context api for now if you are willing to explore, else passing the props should do :)

Reply:
I'm relatively new to TypeScript and haven't worked with TypeScript + React before. As a result, I couldn't complete the task when you mentioned earlier. Could we open a new refactoring issue to address this after the mege, please lmk your thoughts on this.